### ISSUE TITLE
It is too problematic to program bots to attack.

### ISSUE THREAD

**Issue by LucasGerosa** [Adam13531/BotLand#240]
**Steps to reproduce**
Try to edit attack code.
**Expected outcome**
There should be a way to edit attack codes without having to attack someone first;
There should be a way to there should be a way to save your code before having to put your bots and complete a round of attack;
There should be a kind of link, downloadable file,  or a text that you can copy, to save your programming in case of the site crashing or in case of you having to leave for some reason and not have time to complete the attack. 
 Maybe an Auto Save would be useful, for e.g.: The game crashes while you were working for 4 hours on a program. Your only option is to reload the page, but with auto save it could ask: "You did not save, do you want to save it now?" This would save at least until the last code you did while online.

**Actual outcome**
Hours of effort lost because of programs being deleted. ¬¬
**Browser (and version)**

**Operating system (and version)**
****

**Comment by LucasGerosa** [comment]
Also, another thing would be shared scripts with defense and attack, or a way to copy the script from one to another. That would be really useful when you want to make similar bots for defense and attack.

**Comment by Pnoexz** [comment]
I've been programming using my favorite text editor and simply copypasting into BotLand. This allows me to use git to track my changes and will allow me to restore my scripts in case there is a problem with the website.

As for testing your attack before actually attacking somebody, you can edit your defense and enter full test mode. This will allow you to make any changes to your scripts.
![image](https://user-images.githubusercontent.com/3297901/42295431-dc3c9482-7fd9-11e8-9018-1447293cd10d.png)
![image](https://user-images.githubusercontent.com/3297901/42295436-e8122a60-7fd9-11e8-8cae-92db7371a82e.png)
![image](https://user-images.githubusercontent.com/3297901/42295440-f1d47ee0-7fd9-11e8-9ed3-bd1a925e4972.png)
![image](https://user-images.githubusercontent.com/3297901/42295445-fa968456-7fd9-11e8-92c8-46a1af8bae1e.png)

**Comment by HiDeoo** [comment]
> I've been programming using my favorite text editor and simply copypasting into BotLand. This allows me to use git to track my changes and will allow me to restore my scripts in case there is a problem with the website.

Even if this help for some issues in some cases, this is unfortunately not something you can ask of any common user and also make you lose some features.

**Comment by LucasGerosa** [comment]
> As for testing your attack before actually attacking somebody, you can edit your defense and enter full test mode. This will allow you to make any changes to your attack scripts.

Thank you for your suggestion, I did not know that, but that is still a little inconvenient just to edit your code, so having an edit program button on the initial screen would be much more practical.

**Comment by Pnoexz** [comment]
I agree, it's a little annoying but at least it's a way to test your attacks without having to actually attack other people. You can also do it while an actual attack is in progress, which is nice if you need to change your script on the fly.
